synth.py

```diff
@@ -19,39 +19,56 @@
 from synthtool import gcp
 from synthtool.languages import python
 
-gapic = gcp.GAPICBazel()
+gapic = gcp.GAPICMicrogenerator()
 common = gcp.CommonTemplates()
 versions = ["v1beta1", "v1"]
 
 # ----------------------------------------------------------------------------
 # Generate texttospeech GAPIC layer
 # ----------------------------------------------------------------------------
 for version in versions:
-    library = gapic.py_library(
-        service="texttospeech",
-        version=version,
-        bazel_target=f"//google/cloud/texttospeech/{version}:texttospeech-{version}-py",
-        include_protos=True,
-    )
-    s.move(library / f"google/cloud/texttospeech_{version}")
-    s.move(library / f"tests/unit/gapic/{version}")
-    s.move(library / f"docs/gapic/{version}")
-
-# Use the highest version library to generate import alias.
-s.move(library / "google/cloud/texttospeech.py")
+    library = gapic.py_library(service="texttospeech", version=version,)
+    s.move(library, excludes=["setup.py", "docs/index.rst"])
 
 # Fix bad docstrings.
 s.replace("**/gapic/*_client.py", r'\\"(.+?)-\*\\"', r'"\1-\\*"')
 
+# Sphinx interprets `*` as emphasis
+s.replace(
+    ["google/cloud/**/client.py", "google/cloud/**/cloud_tts.py"],
+    "((en)|(no)|(nb)(cmn)|(yue))-\*",
+    "\g<1>-\*",
+)
+
 # ----------------------------------------------------------------------------
 # Add templated files
 # ----------------------------------------------------------------------------
-templated_files = common.py_library(unit_cov_level=85, cov_level=85)
+templated_files = common.py_library(unit_cov_level=85, cov_level=85, samples=True)
 s.move(templated_files)
 
+# Modifications for microgenerator code
+s.replace("noxfile.py",
+"""if os\.path\.exists\("samples"\):
+    BLACK_PATHS\.append\("samples"\)""",
+"")
+
+s.replace("noxfile.py",
+"""python=\["2\.7", """,
+"""python=[""")
+
+s.replace("noxfile.py",
+"""python=\["3\.5", """,
+"""python=[""")
+
```

Comment:
Can you explain a little what's going on here? Is this a lacuna or an error in the generated surface?

Reply:
No, this is is just a temporary hack for the synthtool templates (this noxfile template is used by all the libraries so I can't drop the older Python versions from the template source).

I'll work on a proper fix that lets these lines be added/removed depending on kwargs to the synthtool templates.

Reply:
Removed